site/src/components/MdxPageContent.tsx

```diff
@@ -22,50 +22,71 @@ export const MdxPageContent: VFC<MdxPageContentProps> = ({
 
   const [headings, setHeadings] = useState<Heading[]>([]);
 
-  useEffect(() => {
-    setHeadings([]);
-  }, [serializedPage]);
-
-  const [currentHeading, setCurrentHeading] = useState<Heading>();
   const [detectHeadingFromScroll, setDetectHeadingFromScroll] =
     useState<boolean>(true);
 
-  const [anchor, setAnchor] = useState<string>(
-    router.asPath.split("#")[1] ?? "",
-  );
+  const currentHeading = useRef<Heading | undefined>(undefined);
 
   useEffect(() => {
-    setAnchor(router.asPath.split("#")[1] ?? "");
-  }, [router]);
+    setHeadings([]);
 
-  useEffect(() => {
-    const headingWithCurrentAnchor = headings.find(
-      (heading) => heading.anchor === anchor,
```

Comment:
The inconsistency in scrolling behavior was coming from this line. Since the anchor of the first element always remains the same (#), this would mean that the scroll is never triggered if the user is already at the first anchor of a different spec/docs page.